server/sonar-server/src/test/java/org/sonar/server/computation/queue/ReportSubmitterTest.java

```diff
@@ -113,6 +113,7 @@ public void provision_project_if_does_not_exist() throws Exception {
     when(componentService.create(any(DbSession.class), any(NewComponent.class))).thenReturn(createdProject);
     when(permissionTemplateService.wouldUserHavePermissionWithDefaultTemplate(any(DbSession.class), anyLong(), eq(SCAN_EXECUTION), anyString(), eq(PROJECT_KEY), eq(Qualifiers.PROJECT)))
       .thenReturn(true);
+    when(permissionTemplateService.hasProjectCreator(any(DbSession.class), any(ComponentDto.class))).thenReturn(true);
```

Comment:
In the same way an assertion is missing to verify that favorite is not created when submitting a report on an existing project.